src/Orchard.Web/Modules/Orchard.Users/Module.cs

```diff
@@ -29,11 +30,12 @@ public Startup(ShellSettings shellSettings, IOptions<IdentityOptions> options)
 
         public override void Configure(IApplicationBuilder builder, IRouteBuilder routes, IServiceProvider serviceProvider)
         {
+            builder.UseIdentity();
```

Comment:
`UseIdentity()` doesn't register services. It only registers the cookies middleware in the ASP.NET Core pipeline, which is already what this code block does. If it doesn't work without `UseIdentity()`, I'd be tempted to say there's a bug in how you isolate sub-pipelines.

Do `builder.ApplicationServices` and `serviceProvider` return the same instance?